src/textual/widgets/_data_table.py

```diff
@@ -1308,10 +1355,6 @@ def _update_dimensions(self, new_rows: Iterable[RowKey]) -> None:
         # so that we can cache this rendering for later.
         if auto_height_rows:
             render_cell = self._render_cell  # This method renders & caches.
-            should_highlight = self._should_highlight
-            cursor_type = self.cursor_type
-            cursor_location = self.cursor_coordinate
-            hover_location = self.hover_coordinate
```

Comment:
Okay thanks for the heads up.
I isolated this refactor to 43547603b75fc7a54b455dd0e6972aa6976ef9a5, so it should be reasonably easy to test performance before/after. I'm curious how much of a different it will make.

To be sure: I think I'm doing the same number of comparisons, it's just e.g. `self.cursor_type == "row"` instead of `type_of_cursor == "row"`.